src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -697,5 +718,55 @@ public void Dispose()
                 IOExtensions.DeleteFile(_tempFile);
             }
         }
+
+        public static bool ShouldResolveAttachmentsConflict(BlittableJsonReaderObject myMetadata, BlittableJsonReaderObject otherMetadata)
+        {
+            myMetadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray myAttachments);
+            otherMetadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray otherAttachments);
+            Debug.Assert(myAttachments != null || otherAttachments != null, "Cannot happen. We verified that we have a conflict in @attachments.");
+
+            var myAttachmentNames = new Dictionary<string, BlittableJsonReaderObject>();
+            if (myAttachments != null)
+            {
+                foreach (BlittableJsonReaderObject attachment in myAttachments)
+                {
+                    if (attachment.TryGet(nameof(AttachmentResult.Name), out string name) == false)
+                        Debug.Assert(false, "Cannot happen. Attachment must have a name.");
+
+                    if (myAttachmentNames.ContainsKey(name))
+                        // The node itself has a conflict
+                        return false;
+                    myAttachmentNames.Add(name, attachment);
+                }
+            }
+
+            var otherAttachmentNames = new Dictionary<string, BlittableJsonReaderObject>();
+            if (otherAttachments != null)
+            {
+                foreach (BlittableJsonReaderObject attachment in otherAttachments)
+                {
+                    if (attachment.TryGet(nameof(AttachmentResult.Name), out string name) == false)
+                        Debug.Assert(false, "Cannot happen. Attachment must have a name.");
+
+                    if (otherAttachmentNames.ContainsKey(name))
+                        // The node itself has a conflict
+                        return false;
+                    otherAttachmentNames.Add(name, attachment);
+                }
+            }
+
+            foreach (var attachment in myAttachmentNames)
+            {
+                if (otherAttachmentNames.TryGetValue(attachment.Key, out var otherAttachment))
+                {
+                    if (Document.ComparePropertiesExceptionStartingWithAt(attachment.Value, otherAttachment) == DocumentCompareResult.DifferenceDetected)
+                        return false;
+
+                    otherAttachmentNames.Remove(attachment.Key);
+                }
+            }
+            
+            return true;
```

Comment:
If we don't have the same attachment name than we'll end up with 5 attachments. 
Or if there is an equal name but no conflict in the attachments hash, content type and length than it will merge them.
Otherwise, we will end up with a conflict.
The code that actually merge the attachment is the Put method.